src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5250,6 +5250,17 @@ public override BoundNode VisitDelegateCreationExpression(BoundDelegateCreationE
                         }
                     }
                     break;
+                case BoundExpression arg when arg.Type is { TypeKind: TypeKind.Delegate } argType:
+                    {
+                        var argTypeWithAnnotations = TypeWithAnnotations.Create(argType, NullableAnnotation.NotAnnotated);
+                        var argState = VisitRvalueWithState(arg);
+                        ReportNullableAssignmentIfNecessary(arg, argTypeWithAnnotations, argState, useLegacyWarnings: false);
+                        if (!arg.IsSuppressed)
+                        {
+                            ReportNullabilityMismatchWithTargetDelegate(arg.Syntax.Location, delegateType, arg.Type.DelegateInvokeMethod(), invokedAsExtensionMethod: false);
+                        }
+                    }
+                    break;
                 default:
                     VisitRvalue(node.Argument);
```

Comment:
>VisitRvalue [](start = 20, length = 11)

It feels like the issue is we're not calling `VisitConversion` here. For instance, we're not reporting any warning for the following:
```C#
delegate void D<T>(T t);
class Program
{
    static void M(D<object> d)
    {
        _ = new D<object?>(d);
    }
}
```
Consider changing this to something like:
```C#
VisitOptionalImplicitConversion(
    node.Argument,
    TypeWithAnnotations.Create(delegateType, NullableAnnotation.NotAnnotated),
    useLegacyWarnings: false,
    trackMembers: false,
    assignmentKind: AssignmentKind.Assignment);
```

Reply:
Actually, calling `VisitConversion` probably won't work because that method does not handle implicit MethodGroup conversions.

---
In reply to: [315370081](https://github.com/dotnet/roslyn/pull/38063#discussion_r315370081) [](ancestors = 315370081)